### DISCUSSION TITLE
A way to diagnose why regexs not working

### DISCUSSION THREAD

**Discussion by caoilte** [mxpv/podsync#266] [General]
I have a title regex that used to work and is now failing. I'm struggling to figure out why. One reason might be because the publisher has recently started putting html in their youtube video titles eg

https://www.youtube.com/watch?v=tnI3y8jSebw

`"\\| \\#TyskySour"` used to work. It now doesn't. It might be a problem with the way podsync handles regexs on html, or it might be I need to adjust my regex to handle the html but it's hard to tell as podsync doesn't log the text of the title that it failed to match. Would it be reasonable to change that? 

eg

```
time="2021-01-07T19:59:23Z" level=info msg="skipping due to mismatch" episode_id=i5KTGbiY68U filter=title
```

could become

```
time="2021-01-07T19:59:23Z" level=info msg="skipping due to mismatch on title 'Chris Whitty Warns Of Covid Restrictions NEXT Winter | #TyskySour"  episode_id=i5KTGbiY68U filter=title
```

Even then it might be hard to spot the character encoding problems, however.

**Comment by Rpsl** [comment]
I think problem not with html.

I try made API request with [google api console](https://developers.google.com/youtube/v3/docs/videos/list) and answer dont't have any kind of html

```json
  "etag": "...",
  "items": [
    {
      "kind": "youtube#video",
      "etag": "...",
      "id": "tnI3y8jSebw",
      "snippet": {
        "publishedAt": "2021-01-06T20:40:40Z",
        "channelId": "UCOzMAa6IhV6uwYQATYG_2kg",
        "title": "Chris Whitty Warns Of Covid Restrictions NEXT Winter | #TyskySour",
        "description": ....
```

You can test your regexp online via https://regex101.com/r/n2CZom/1.
